pilot/pkg/networking/core/v1alpha3/gateway.go

```diff
@@ -196,8 +206,85 @@ func (configgen *ConfigGeneratorImpl) buildGatewayHTTPRouteConfig(env *model.Env
 		nameToServiceMap[svc.Hostname] = svc
 	}
 
-	routeCfg := configgen.buildGatewayInboundHTTPRouteConfig(env, node, nameToServiceMap, merged.Names, servers, routeName)
-	log.Debugf("Returning route config (%s) %v", routeName, routeCfg)
+	gatewayHosts := make(map[model.Hostname]bool)
```

Comment:
This is simply collapsing the two functions. The other one was a leftover from before RDS. And the names are super confusing.